django/db/models/sql/query.py

```diff
@@ -855,8 +855,10 @@ def change_aliases(self, change_map):
                 if alias == old_alias:
                     table_aliases[pos] = new_alias
                     break
-        self.external_aliases = {change_map.get(alias, alias)
-                                 for alias in self.external_aliases}
+        self.external_aliases = {
+            change_map.get(alias, alias): (aliased or alias in change_map)
+            for alias, aliased in self.external_aliases.items()
```

Comment:
I had to read it 3 times to understand it, maybe the wording could be improved here. Alias and aliased is just very similar.

Reply:
TBH, I couldn't find a better wording.